### DISCUSSION TITLE
MPC Autotune not working for me

### DISCUSSION THREAD

**Discussion by krak3n1013** [mriscoc/Ender3V2S1#599] [Q&A]
Trying to figure out this MPC Autotune on my Ender 3 v2 w/ Sprite Pro, using firmware "Ender3V2-422-BLTUBL-T13-MPC". 
When I ran the autotune test, the test worked great, heated nozzle to 209c and the test did it's thing. Once test was done I tried heating nozzle to my operating print temp, the heater shot past the my set print temp and kept on climbing until I turned my printer off. I then turned my printer back on and cooled it down some and then tried heating up my nozzle again, set nozzle temp to my print temp and again the heater shot past my set temp and kept on climbing.  Also it doesn't matter what temp I set, I tried at 10c and at 100c, at both those temps the nozzle heater just climbed in temp until I stopped it. What can I do to fix this?

**Comment by mriscoc** [comment]
Did you save the MPC parameters after the test?

**Comment by krak3n1013** [comment]
I'm pretty sure I saved once tune was done. Going to download and try installing T13-MPC firmware again.

**Reply by mriscoc** [reply]
Please verify the version of your board also.

**Reply by krak3n1013** [reply]
Board is an V4.2.2, STM32103, Ret6

So here is what I just did. I downloaded "Ender3V2-422-BLTUBL-T13-MPC", installed on my printer. I then went straight to the MPC Autotune and ran the test. Once test was complete I backed out and stored settings, I then went to prepare and clicked on the preheat Pla. The heaters came on and the nozzle started to heat up and only heated to 188c, then the nozzle heater started to cool down, bed temp is at 60c and holding. A message came up on the LCD "Nozzle or Bed temp to low" then another message "Printer Killed! Heating Failed Turn off Printer.

**Reply by mriscoc** [reply]
OK thanks, I will try to check.

**Reply by mriscoc** [reply]
I have an almost stock Ender3V2 so no T13, but since it's just a test, I overlooked the imprecision of the thermistor type.
I loaded the firmware Ender3V2-422-BLTUBL-T13-MPC, after reboot I ran the MPC Autotune, saved the parameters and went to Prepare/Filament management and Preheat for PLA, the temperature was ok and stable.

**Reply by krak3n1013** [reply]
Thanks for checking. I haven't given up just yet on the T-13 MPC firmware. I do have a question when you saved the MPC parameters, did you just back out of the MPC tune screen and scroll up and hit the "Store Settings" tab? I'm only asking to verify that I saved the tune correctly.

**Reply by mriscoc** [reply]
Yes, I just go back to Advanced Settings menu and save settings

**Comment by JovelVasquez** [comment]
Hey @krak3n1013, I have the same exact issue did you figured it out?  

I wanted to take a moment to express my sincerest gratitude for the remarkable work you have put into developing the firmware @mriscoc

**Reply by JovelVasquez** [reply]
FIXED - Update! After replacing the heating block with a new one, I conducted measurements on the heating resistors and found that the values are consistent in both units. However, I did notice a discrepancy of 15 ohms in the temperature sensor. It is possible that this difference could be causing the problem, also notice the little cables was kind of damage near the sensor. The printer was reading the temp fine but the MPC Autotuned was not working properly  👍

**Comment by Amesis2077** [comment]
Ender3V2-422-MM-MPC-20260106.bin has excellent gui and features but I have Ender 3 v2 v4.2.2 stm32f103 ret6+dwin with silent motors, when i load firmware [Ender3V2-422-MM-MPC-20260106.bin] display is ok limit switches are ok, bed heating is ok but when i select autohome or try to print sth, motors lock, hotend doesnt respond and after a while machine is killed warning signs until reboot i tried everything renamed firmware, tried to delete mainboards mem, but the result was same how can i solve the problem
